add favourite confirmation modal

chriswmartin opened the pull request:
I honestly don't know if this is something that anyone else wants besides me (I can't find any feature requests for it on the tootsuite repo) but I've been wanting the option to have a favourite confirmation modal for a while so here it is

wip because I should probably move the toggle for it out of the vanilla preferences and into the glitch app settings modal and I'm not sure if its bad glitch form to add to:
```
app/controllers/settings/preferences_controller.rb
app/models/user.rb
app/lib/user_settings_decorator.rb
```

marrus-sh commented:
I would say leave it in the vanilla preferences actually; eventually we'll build API points and such but until then keeping it with the reblog/delete confirmation modals is probably a good thing. I haven't taken too close of a look at the code yet but the approach seems right to me.

chriswmartin commented:
okay cool, yeah I feel like it makes the most sense for it to be located next to the reblog/delete options I just wasn't sure how strictly glitch features should be kept separate

I'll revert my app settings modal additions

marrus-sh commented:
there aren't any hard guidelines about changing the ruby/backend files, but you're mostly just adding lines so it should be okay :)

aside from wanting to avoid changes that will cause difficult merge conflicts in the future, we're pretty relaxed about things i think

chriswmartin commented:
I reverted the app settings additions and removed the WIP label from this PR. I feel pretty good about this now other than the fact that the "Show confirmation dialog before favouriting" check box still shows up in vanilla preferences even though it only has an effect when using a glitch flavour. I think that's probably not a big deal though.

marrus-sh commented on app/javascript/flavours/glitch/features/ui/components/actions_modal.js:
```diff
@@ -56,6 +56,22 @@ export default class ActionsModal extends ImmutablePureComponent {
+            <DisplayName account={this.props.status.get('account')} />
+          </a>
+        </div>
+
```
This won't ever render because `actions_modal` is never called with a `status` prop (I'm… not entirely sure why it accepts one lol) but you don't need to duplicate this code—this would cause the status header to be rendered twice.

chriswmartin commented:
thanks, I'm not sure how I missed that
